diff --git a/src/maasserver/preseed_network.py b/src/maasserver/preseed_network.py
index 1203e6b..5475b18 100644
--- a/src/maasserver/preseed_network.py
+++ b/src/maasserver/preseed_network.py
@@ -307,7 +307,11 @@ class InterfaceConfiguration:

     def _get_matching_routes(self, source):
         """Return all route objects matching `source`."""
-        return {route for route in self.routes if route.source == source}
+        return {
+            route
+            for route in self.routes
+            if str(route.source.cidr) == str(source.cidr)
+        }

     def _generate_addresses(self):
         """Generate the various addresses needed for this interface."""
