diff --git a/src/provisioningserver/utils/network.py b/src/provisioningserver/utils/network.py
index 7895227c4..df83836f3 100644
--- a/src/provisioningserver/utils/network.py
+++ b/src/provisioningserver/utils/network.py
@@ -1128,6 +1128,7 @@ def get_all_interfaces_definition(
         # interfaces for guests. By themselves, they're not useful for MAAS to
         # manage.
         "tunnel",
+        "ethernet",
     ]
     if not running_in_container():
         # When not running in a container, we should be able to identify
