diff --git a/src/maasserver/compose_preseed.py b/src/maasserver/compose_preseed.py
index f24b10f..f030469 100644
--- a/src/maasserver/compose_preseed.py
+++ b/src/maasserver/compose_preseed.py
@@ -8,7 +8,7 @@ __all__ = [
     ]

 from datetime import timedelta
-from urllib.parse import urlencode
+from urllib.parse import urlencode, urlparse

 from maasserver.clusterrpc.osystems import get_preseed_data
 from maasserver.enum import (
@@ -17,6 +17,7 @@ from maasserver.enum import (
     PRESEED_TYPE,
 )
 from maasserver.models import PackageRepository
+from maasserver.config import RegionConfiguration
 from maasserver.models.config import Config
 from maasserver.server_address import get_maas_facing_server_host
 from maasserver.utils import absolute_reverse
@@ -43,11 +44,11 @@ def get_apt_proxy(rack_controller=None, default_region_ip=None):
         else:
             maas_proxy_port = Config.objects.get_config("maas_proxy_port")
             if not maas_proxy_port:
-                maas_proxy_port = 8000
+                maas_proxy_port = 31800
             url = "http://:%d/" % maas_proxy_port
-            return compose_URL(
-                url, get_maas_facing_server_host(
-                    rack_controller, default_region_ip=default_region_ip))
+            with RegionConfiguration.open() as config:
+                maas_url = config.maas_url
+            return compose_URL(url, urlparse(maas_url).hostname)
     else:
         return None
