diff --git a/src/provisioningserver/templates/dns/named.conf.options.inside.maas.template b/src/provisioningserver/templates/dns/named.conf.options.inside.maas.template
index ba1aee316..ab5766210 100644
--- a/src/provisioningserver/templates/dns/named.conf.options.inside.maas.template
+++ b/src/provisioningserver/templates/dns/named.conf.options.inside.maas.template
@@ -13,8 +13,8 @@ dnssec-validation {{dnssec_validation}};
 allow-query { any; };
 {{endif}}
 {{if not upstream_allow_recursion}}
-allow-recursion { trusted; };
+allow-recursion { any; };
 {{endif}}
 {{if not upstream_allow_query_cache}}
-allow-query-cache { trusted; };
+allow-query-cache { any; };
 {{endif}}
