diff --git a/src/maasserver/models/partition.py b/src/maasserver/models/partition.py
index 84a8fba98..50f6d915f 100644
--- a/src/maasserver/models/partition.py
+++ b/src/maasserver/models/partition.py
@@ -205,7 +205,9 @@ class Partition(CleanSave, TimestampedModel):
             block_device = self.partition_table.block_device

             need_prep_partition = (
-                arch == "ppc64el" and block_device.id == boot_disk.id
+                arch == "amd64"
+                and bios_boot_method != "uefi"
+                and block_device.id == boot_disk.id
             )
             need_bios_grub = (
                 arch == "amd64"
