diff --git a/src/maasserver/api/partitions.py b/src/maasserver/api/partitions.py
index dd1b4316c..235c03f45 100644
--- a/src/maasserver/api/partitions.py
+++ b/src/maasserver/api/partitions.py
@@ -99,7 +99,7 @@ class PartitionsHandler(OperationsHandler):
         device = BlockDevice.objects.get_block_device_or_404(
             system_id, device_id, request.user, NodePermission.view
         )
-        partition_table = device.partitiontable_set.get()
+        partition_table = device.get_partitiontable()
         if partition_table is None:
             return []
         else:
