diff --git a/src/provisioningserver/utils/network.py b/src/provisioningserver/utils/network.py
index 6124f68..fb1fa0b 100644
--- a/src/provisioningserver/utils/network.py
+++ b/src/provisioningserver/utils/network.py
@@ -1187,6 +1187,8 @@ def get_all_interfaces_definition(
         # interfaces for guests. By themselves, they're not useful for MAAS to
         # manage.
         "tunnel",
+        # Always exclude non-specific ethernet interfaces.
+        "ethernet",
     ]
     if not running_in_container():
         # When not running in a container, we should be able to identify
